prefect_kubernetes/jobs.py

```diff
@@ -299,3 +309,197 @@ def kubernetes_orchestrator():
+                    label_selector=(
+                        f"controller-uid="
+                        f"{latest_v1_job.metadata.labels['controller-uid']}"
+                    ),
+                    **self._kubernetes_job.api_kwargs,
+                )
+                for pod in v1_pod_list.items:
+                    pod_name = pod.metadata.name
+
+                    if pod.status.phase == "Pending" or pod_name in self.pod_logs:
+                        continue
+
+                    self.logger.info(f"Capturing logs for pod {pod_name}.")
+
+                    self.pod_logs[pod_name] = await run_sync_in_worker_thread(
+                        core_v1_client.read_namespaced_pod_log,
+                        namespace=self._kubernetes_job.namespace,
+                        name=pod_name,
+                        container=latest_v1_job.spec.template.spec.containers[0].name,
+                        **self._kubernetes_job.api_kwargs,
+                    )
+
+                if latest_v1_job.status.active:
+                    await sleep(self._kubernetes_job.interval_seconds)
+                elif latest_v1_job.status.failed:
+                    raise RuntimeError(
+                        f"Job {latest_v1_job.metadata.name} failed, check the "
+                        "Kubernetes pod logs for more information."
+                    )
+                elif latest_v1_job.status.succeeded:
+                    completed = True
+                    self.logger.info(
+                        f"Job {latest_v1_job.metadata.name} has completed."
+                    )
+
+    async def fetch_result(self) -> Dict[str, Any]:
+        """Fetch the results of the job.
+
+        Returns:
+            The logs from each of the pods in the job.
+        """
+        if self._kubernetes_job.delete_after_completion:
+            with self._kubernetes_job.credentials.get_client(
+                "batch"
+            ) as batch_v1_client:
+                deleted_v1_job = await run_sync_in_worker_thread(
+                    batch_v1_client.delete_namespaced_job,
+                    namespace=self._kubernetes_job.namespace,
+                    name=self._kubernetes_job.v1_job.metadata.name,
+                    **self._kubernetes_job.api_kwargs,
+                )
+                self.logger.info(
+                    f"Job {self._kubernetes_job.v1_job.metadata.name} deleted "
+                    f"with {deleted_v1_job.status!r}."
+                )
+
+        return self.pod_logs
+
+
+class KubernetesJob(JobBlock):
+    """A block representing a Kubernetes job configuration."""
+
+    api_kwargs: Optional[Dict[str, Any]] = Field(
+        default_factory=dict,
+        description="The kwargs to pass to all Kubernetes API calls.",
+        example={"pretty": "true"},
+    )
+    credentials: KubernetesCredentials = Field(
+        default=..., description="The credentials to configure a client from."
+    )
```

Comment:
Should credentials be first, but not sure if this matters.

Also, should it be named `kubernetes_credentials`? Not sure if we have a standard yet though

Lastly, does Credentials work with `Field` on the Blocks UI? I remember some issues with it.

Reply:
personally I like `credentials` since there's the logo and it seems like an easy way to standardize something, i imagine it will be a familiar kwarg among our job / abstract blocks in general

but i could understand wanting to be more explicit for the UI field's sake

![image](https://user-images.githubusercontent.com/31014960/208844437-d907c618-26cd-4096-849f-254d17049737.png)